sklearn/isotonic.py

```diff
@@ -121,10 +122,7 @@ def isotonic_regression(y, sample_weight=None, y_min=None, y_max=None,
     order = np.s_[:] if increasing else np.s_[::-1]
     y = check_array(y, ensure_2d=False, dtype=[np.float64, np.float32])
     y = np.array(y[order], dtype=y.dtype)
-    if sample_weight is None:
-        sample_weight = np.ones(len(y), dtype=y.dtype)
-    else:
-        sample_weight = np.array(sample_weight[order], dtype=y.dtype)
+    sample_weight = _check_sample_weight(sample_weight, y, dtype=y.dtype)
```

Comment:
Thanks!

You need to apply the `order` parameter, maybe as,
```py
    if sample_weight is None:
        sample_weight = np.ones(len(y), dtype=y.dtype)
    else:
        sample_weight = _check_sample_weight(sample_weight, y, dtype=y.dtype)
        sample_weight = sample_weight[order]
```

Reply:
don't you think
```
sample_weight = _check_sample_weight(sample_weight, y, dtype=y.dtype)
sample_weight = sample_weight[order]
```
is enough ? It makes the code simpler and reordering ones has no effect and efficiency is not a problem there.

Reply:
> It makes the code simpler and reordering ones has no effect and efficiency is not a problem there.

If you prefer. Slicing an array with int indexing does have some cost, but it should indeed be minimal.

Reply:
I thought the same :)